crates/editor/src/editor_tests.rs

```diff
@@ -26962,6 +26962,43 @@ async fn test_copy_line_without_trailing_newline(cx: &mut TestAppContext) {
     cx.assert_editor_state("line1\nline2\nˇ");
 }
 
+#[gpui::test]
+async fn test_multi_selection_copy_with_newline_between_copied_lines(cx: &mut TestAppContext) {
+    init_test(cx, |_| {});
+
+    let mut cx = EditorTestContext::new(cx).await;
+
+    cx.set_state("line1\nline2\nline3\nˇ");
+
+    cx.update_editor(|e, window, cx| {
```

Comment:
This all is not needed, as we can place multiple `ˇ` markers into the input text for `set_state` with the same result.

Same works in `assert_editor_state`.